src/odemis/odemisd/start.py

```diff
@@ -722,8 +722,28 @@ def main(args):
             else:
                 # Return when the GUI is done
                 logging.info("Starting the GUI...")
-                subprocess.check_call(["odemis-gui", "--log-level", odemis_config["LOGLEVEL"]])
 
+                class GuiError(Exception):
+                    pass
+
+                try:
+                    subprocess.check_call(["odemis-gui", "--log-level", odemis_config["LOGLEVEL"]])
```

Comment:
The "problem" with `check_call` is that it waits until the process ends (to get the exit code). So if the GUI starts fine, it will stay waiting until the user ends the GUI. So, I think that typically `find_window(GUI_WM_CLASS)` will never find anything.
The additional problem is that for now when the user selects "Full stop" of Odemis, the GUI is just killed, and so it returns a error code. In this case, the code here would indicate that the GUI failed. Admittedly, this second issue could be solved by asking more kindly the GUI to close (eg, `wmctrl -c {GUI_WM_CLASS} -x`).

To check that the error really happened at the beginning, I'd add a condition on the time it took to run. For instance, if an error happened and it ran for less than 10s, it is considered a "GUI start error", anything else is fine.